client/tracing/src/lib.rs

```diff
@@ -456,7 +455,7 @@ mod tests {
 			Box::new(handler),
 			"test_target",
 		);
-		let subscriber = tracing_subscriber::fmt().finish().with(layer);
+		let subscriber = tracing_subscriber::fmt().with_writer(std::io::sink).finish().with(layer);
```

Comment:
Let me se if I understand this correctly: instead of having our own `TestSubscriber`, you create one that writes to what is essentially `/dev/null` (will always succeed) and access the spans/events through the Arc'd (like before).

That seems like a much better solution.

Reply:
Yeah, I first thought that this would be one of the problems. In the end this is no problem at all, but should make the output cleaner.